### PR TITLE
updated InteractableInspector to fix mismatch repaint issue

### PR DESCRIPTION
**Pull request by killerantz**
## Overview
Fix for Issue #4011 

When initializing the Interactable the inspector does a check to see if the States property is not null. If it's null, then we try to fill it with the DefaultInteractableStates asset.

The issue came up when the States property was filled in the layout phase, but never rendered. During the repaint phase, now that the property is not null, we attempt to render the field in the inspector.

## Changes
We are now rendering the field in both phases, in the layout phase after the property is filled.

## Verification
I was able to consistently repro this by adding the Interactable component to a GameObject using the Add Component button in the inspector.

1 Create a new Scene
2 Add MRTK by selecting MRTK/Add to Scene and Configure... in the menu
3 Drag a pressable button into the scene
4 press play
5 observe no errors

or 

1 Create a game object in the scene
2 Select it and press Add Component in the inspector
3 Select Interactable as the component to add
4 observe no errors.

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs [hunk 1]
```diff
@@ -135,7 +139,7 @@ public virtual void RenderCustomInspector()
             {
                 InspectorUIUtility.DrawSectionEnd(indentOnSectionStart);
             }
-
+
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs]:
```suggestion

```

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs [hunk 2]
```diff
@@ -89,16 +89,17 @@ public virtual void RenderCustomInspector()
...
             {
-                string statesPrefKey = "Settings_States";
-                bool prefsShowStates = EditorPrefs.GetBool(statesPrefKey);
                 EditorGUI.indentLevel = indentOnSectionStart + 1;
                 showStates = InspectorUIUtility.DrawSectionStart(states.objectReferenceValue.name + " (Click to edit)", indentOnSectionStart + 2, prefsShowStates, FontStyle.Normal, false);
                 drawerStarted = true;
                 if (showStates != prefsShowStates)
                 {
-                    EditorPrefs.SetBool(statesPrefKey, showStates);
+                   EditorPrefs.SetBool(statesPrefKey, showStates);
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs]:
```suggestion
                    EditorPrefs.SetBool(statesPrefKey, showStates);
```

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs [hunk 3]
```diff
@@ -89,16 +89,17 @@ public virtual void RenderCustomInspector()
             bool showStates = false;
             SerializedProperty states = serializedObject.FindProperty("States");
             bool drawerStarted = false;
+            string statesPrefKey = "Settings_States";
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs]: Was it necessary to bring out these two statements into this scope? It looks like these are still only used inside the if clause of line 95

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs [hunk 4]
```diff
@@ -89,16 +89,17 @@ public virtual void RenderCustomInspector()
...
+            bool prefsShowStates = EditorPrefs.GetBool(statesPrefKey);
+
             if (states.objectReferenceValue != null)
             {
-                string statesPrefKey = "Settings_States";
-                bool prefsShowStates = EditorPrefs.GetBool(statesPrefKey);
                 EditorGUI.indentLevel = indentOnSectionStart + 1;
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs]: Can you elaborate a bit more (on the PR, maybe not in comments in the code) about why an indent happens here but not down below?

Comment (killerantz) [reply Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs]: That's an oversight. I'll update it.

**Comment by julenka** [comment]
uh oh, looks like boundingbox might have an unreliable test.

**Pull request merged**